libpod/boltdb_state_internal.go

```diff
@@ -137,6 +137,9 @@ func validateDBAgainstConfig(bucket *bolt.Bucket, fieldName, runtimeValue string
 				return nil
 			}
 
+			if (strings.EqualFold(fieldName, "graph driver name")) {
+				return errors.New("Configuration specifies " + runtimeValue + " storage driver but detected " + string(keyBytes) + "\nPlease change the configuration or clean up storage.")
+			}
 			return errors.Wrapf(ErrDBBadConfig, "database %s %s does not match our %s %s",
```

Comment:
Could we improve this error message instead? This also happens for a changed runroot directory for example, and the remedy in most cases is going to be delete storage

Reply:
I'm happy to change the error message.  I thought a failure here might be caused by other db data mismatches so zoned in on that particular case.  Plus I was trying to emulate as much as I can the suggested message from the issue.  If you've a thought on a better error message, I'm happy to take them.

Reply:
@mheon thoughts?

Reply:
The error messages out of here should be improved since the PR to use fields from the database where possible - I think we're in better shape here, aside from maybe needing a note that the configuration files should be altered to fix this.

We also shouldn't see this message almost ever, since we're now sourcing graph driver from the database when possible